src/main/java/hudson/tasks/Mailer.java

```diff
@@ -396,18 +396,19 @@ protected PasswordAuthentication getPasswordAuthentication() {
         public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
 
             BulkChange b = new BulkChange(this);
+            // Nested Describable (SMTPAuthentication) is not set to null in case it is not configured.
+            // To mitigate that, it is being set to null before (so it gets set to sent value or null correctly) and, in
+            // case of failure to databind, it gets reverted to previous value.
+            // Would not be necessary by https://github.com/jenkinsci/jenkins/pull/3669
```

Comment:
I did my best to describe what I going on and why. Feel free o suggest version that is clearer.